src/java.base/share/classes/jdk/internal/misc/CDS.java

```diff
@@ -26,6 +26,15 @@
 package jdk.internal.misc;
 
 public class CDS {
+    // cache the result
+    static private boolean isDumpLoadedClassList;
```

Comment:
`isDumpLoadedClassList` is not gramatically correct. Also the field should be final. How about:

```
static final private boolean isDumpingClassList = isDumpingClassList0();
public static boolean isDumpingClassList() {
    return isDumpingClassList;
}
private static boolean isDumpingClassList0();
```